src/mango_native_proc.erl

```diff
@@ -163,14 +165,19 @@ get_text_field_values({Props}, TAcc) when is_list(Props) ->
     get_text_field_values_obj(Props, TAcc, []);
 
 get_text_field_values(Values, TAcc) when is_list(Values) ->
+    IndexArrayLengths = TAcc#tacc.index_array_lengths,
     NewPath = ["[]" | TAcc#tacc.path],
-    NewTAcc = TAcc#tacc{path = NewPath},
-    % We bypass make_text_field and directly call make_text_field_name
-    % because the length field name is not part of the path.
-    LengthFieldName = make_text_field_name(NewTAcc#tacc.path, <<"length">>),
-    LengthField = [{LengthFieldName, <<"length">>, length(Values)}],
-    get_text_field_values_arr(Values, NewTAcc, LengthField);
-
+    NewTAcc = TAcc#tacc{index_array_lengths = IndexArrayLengths, path = NewPath},
```

Comment:
Will change this. Based on my limited understanding of erlang I thought I have to re initialize the values when you create a record as in "NewTAcc = TAcc#tacc{path = NewPath},". I have to do some more reading on how records work. Thanks for the tip.